File: server/src/controllers/adminController.js
import { getDashboardSummary } from "../services/adminService.js";
import { sendSuccess } from "../utils/apiResponse.js";
import { asyncHandler } from "../utils/asyncHandler.js";

export const getAdminDashboard = asyncHandler(async (_req, res) => {
  const data = await getDashboardSummary();
  return sendSuccess(res, "Admin dashboard fetched", data);
});

File: server-spring/src/main/java/com/ecommerce/modules/admin/controller/AdminController.java
package com.ecommerce.modules.admin.controller;

import com.ecommerce.modules.admin.service.AdminService;
import com.ecommerce.shared.response.ApiResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping("/api/admin")
@RequiredArgsConstructor
public class AdminController {

    private final AdminService adminService;

    @GetMapping("/dashboard")
    @PreAuthorize("hasRole('ADMIN')")
    public ResponseEntity<ApiResponse<Map<String, Object>>> getDashboard() {
        Map<String, Object> data = adminService.getDashboardSummary();
        return ResponseEntity.ok(ApiResponse.success("Dashboard data fetched", data));
    }
}
